packages/core/src/navigation-item/NavigationItem.tsx

```diff
@@ -75,34 +128,68 @@ export const NavigationItem = forwardRef<HTMLDivElement, NavigationItemProps>(
       "--saltNavigationItem-level": `${level}`,
     };
 
+    const elementProps = {
+      className: clsx(
+        withBaseName("wrapper"),
+        {
+          [withBaseName("active")]: active || blurActive,
+          [withBaseName("blurActive")]: blurActive,
+          [withBaseName("rootItem")]: level === 0,
+        },
+        withBaseName(orientation)
+      ),
+      children: (
+        <>
+          <span className={withBaseName("label")}>{children}</span>
+          {parent ? (
+            <ExpansionIcon expanded={expanded} orientation={orientation} />
+          ) : null}
+        </>
+      ),
+    };
+    let parentProps: Partial<ButtonHTMLAttributes<HTMLButtonElement>> = {};
+    let linkProps: Partial<AnchorHTMLAttributes<HTMLAnchorElement>> = {};
+    if (parent) {
+      const handleExpand = onExpand
+        ? (event: MouseEvent<HTMLButtonElement>) => {
+            event.stopPropagation();
+            onExpand(event);
+          }
+        : undefined;
+      parentProps = {
+        ...elementProps,
+        "aria-label": "expand",
+        "aria-expanded": expanded,
+        onClick: handleExpand,
+      };
+    } else {
+      linkProps = {
+        ...elementProps,
+        href,
+        "aria-label": "change page",
+        "aria-current": active ? "page" : undefined,
+      };
+    }
+
+    const defaultElementType = parent ? "button" : "a";
+    const renderedElement = createElement(defaultElementType, {
+      active,
+      expanded,
+      parent,
+      level,
+      orientation,
+      render,
+      linkProps: !parent ? linkProps : undefined,
+      parentProps: parent ? parentProps : undefined,
+    });
     return (
       <div
         ref={ref}
         className={clsx(withBaseName(), className)}
         style={style}
         {...rest}
```

Comment:
ideally would not have this wrapping div and then we could pass `...rest` to the render function, but that would apply the props to a different element and be a  breaking change so kept the current setup. 

(Note: This div does not do anything, we can pass the styles to the render function and it would render the same).